runtime-sdk/modules/evm/src/raw_tx.rs

```diff
@@ -0,0 +1,274 @@
+                std::cmp::min(message.max_fee_per_gas, message.max_priority_fee_per_gas);
+            (
+                sig,
+                sig_digest,
+                message.action,
+                message.value,
+                message.input,
+                message.nonce,
+                resolved_gas_price,
+                message.gas_limit,
+            )
+        }
+    };
+    let (method, body) = match eth_action {
+        ethereum::TransactionAction::Call(eth_address) => (
+            "evm.Call",
+            cbor::to_value(types::Call {
+                address: eth_address.into(),
+                value: eth_value.into(),
+                data: eth_input,
+            }),
+        ),
+        ethereum::TransactionAction::Create => (
+            "evm.Create",
+            cbor::to_value(types::Create {
+                value: eth_value.into(),
+                init_code: eth_input,
+            }),
+        ),
+    };
+    let key = sig
+        .recover_verify_key_from_digest(sig_digest)
+        .with_context(|| "recover verify key from digest")?;
+    let nonce: u64 = eth_nonce
+        .try_into()
+        .map_err(|e| anyhow!("converting nonce: {}", e))?;
+    let gas_price: u128 = eth_gas_price
+        .try_into()
+        .map_err(|e| anyhow!("converting gas price: {}", e))?;
+    let gas_limit: u64 = eth_gas_limit
+        .try_into()
+        .map_err(|e| anyhow!("converting gas limit: {}", e))?;
+    let resolved_fee_amount = gas_price
+        .checked_mul(gas_limit as u128)
+        .ok_or_else(|| anyhow!("computing total fee amount"))?;
+    Ok(transaction::Transaction {
+        version: transaction::LATEST_TRANSACTION_VERSION,
+        call: transaction::Call {
+            format: transaction::CallFormat::Plain,
+            method: method.to_owned(),
+            body,
+        },
+        auth_info: transaction::AuthInfo {
+            signer_info: vec![transaction::SignerInfo {
+                address_spec: transaction::AddressSpec::Signature(signature::PublicKey::Secp256k1(
+                    signature::secp256k1::PublicKey::from_bytes(
+                        k256::EncodedPoint::from(&key).as_bytes(),
+                    )
+                    .with_context(|| "sdk secp256k1 public key from bytes")?,
+                )),
+                nonce,
+            }],
+            fee: transaction::Fee {
+                amount: token::BaseUnits(resolved_fee_amount, token::Denomination::NATIVE),
+                gas: gas_limit,
+                consensus_messages: 0,
+            },
+        },
+    })
+}
+
+#[cfg(test)]
+mod test {
+    use std::str::FromStr as _;
+
+    use oasis_runtime_sdk::types::token;
+
+    use crate::types;
+
+    use super::decode;
+
+    #[test]
+    fn test_decode() {
+        // Some recent transaction seen on Ethereum when we wrote this test.
+        // https://etherscan.io/tx/0x5e5cb07e74593198f66604b0dc4dae6187186c3fa94088f062cc51fc554141de
```

Comment:
Also add tests for the other transaction types, so everything is covered.

Reply:
ya should

Reply:
nvm let's track that here

Reply:
added!